### ISSUE TITLE
Assert failure for pxfprotocol unittest.

### ISSUE THREAD

**Issue by xinzweb** [greenplum-db/gpdb#2964]
When build with `--enable-cassert` the unittest failed:

```
./pxfprotocol.t
[ RUN         ] test_pxfprotocol_validate_urls
[          OK ] test_pxfprotocol_validate_urls
[ RUN         ] test_pxfprotocol_export
[          OK ] test_pxfprotocol_export
[ RUN         ] test_pxfprotocol_import_first_call
FATAL:  (XX000) Unexpected internal error (pxfprotocol.c:126)
          DETAIL:  FailedAssertion("!(context->gphd_uri->fragments != ((void *)0))", File: "../src/pxfprotocol.c", Line: 126)
HINT:  Process 0 will wait for gp_debug_linger=30 seconds before termination.
	Note that its locks and other resources will not be released until then.
LOCATION:  pxfprotocol.c:126
          Traceback 0:  ./pxfprotocol.t() [0x90b80b]
          Traceback 1:  ./pxfprotocol.t() [0x909a6f]
          Traceback 2:  ./pxfprotocol.t() [0xa8c2fc]
          Traceback 3:  ./pxfprotocol.t() [0xa8c1d1]
          Traceback 4:  ./pxfprotocol.t() [0xa8c90c]
          Traceback 5:  ./pxfprotocol.t() [0xa8b7bb]
          Traceback 6:  ./pxfprotocol.t() [0xa8bacb]
          Traceback 7:  ./pxfprotocol.t() [0xa8cfbd]
          Traceback 8:  /lib64/libc.so.6(__libc_start_main+0xfd) [0x7fa1abc92d1d]
          Traceback 9:  ./pxfprotocol.t() [0x409f69]
make[3]: *** [pxfprotocol-check] Error 1
```

**Comment by sansanichfb** [comment]
Looking into it.

**Comment by ashwinstar** [comment]
Fixed by #2976
